src/key.cpp

```diff
@@ -103,51 +107,60 @@ int keybind(int argc, char** argv, Output output) {
     new_bind.cmd_argc = argc - 2;
     new_bind.cmd_argv = argv_duplicate(new_bind.cmd_argc, argv+2);
     // add keybinding
-    KeyBinding* data = g_new(KeyBinding, 1);
-    *data = new_bind;
-    g_key_binds = g_list_append(g_key_binds, data);
+    g_key_binds.push_back(new_bind);
     // grab for events on this keycode
-    grab_keybind(data, nullptr);
+    grab_keybind(&g_key_binds.back());
     return 0;
 }
 
-bool string2modifiers(char* string, unsigned int* modmask) {
+vector<std::string> splitKeySpec(std::string keySpec)
+{
+    // Normalize spec to use a single separator:
+    char baseSep = KEY_COMBI_SEPARATORS[0];
+    for (auto &sep : KEY_COMBI_SEPARATORS) {
+        std::replace(keySpec.begin(), keySpec.end(), sep, baseSep);
+    }
+
+    // Split spec into tokens:
+    vector<std::string> tokens;
+    std::string token;
+    std::istringstream tokenStream(keySpec);
+    while (std::getline(tokenStream, token, baseSep)) {
+        tokens.push_back(token);
```

Comment:
Don't you need to strip off `baseSep` from the end of token?

Reply:
The delimiter gets dropped by `std::getline` ("the delimiter character is extracted from input, but is not appended to str." https://en.cppreference.com/w/cpp/string/basic_string/getline)

Reply:
Thanks!